libs/esp32kernel/esp32kernel.erl

```diff
@@ -0,0 +1,376 @@
+%
+% This file is part of AtomVM.
+%
+% Copyright 2023 Davide Bettio <davide@uninstall.it>
+%
+% Licensed under the Apache License, Version 2.0 (the "License");
+% you may not use this file except in compliance with the License.
+% You may obtain a copy of the License at
+%
+%    http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% See the License for the specific language governing permissions and
+% limitations under the License.
+%
+% SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+%
+
+-module(esp32kernel).
+
+-export([
+    start/0,
+    start_repl/1,
+    start_network/0,
+    maybe_start_network/0,
+    handle_req/3,
+    erase_net_config/0,
+    save_net_config/2
+]).
+
+-record(nc_state, {socket, pending_pid, pending_ref}).
+
+-define(DEFAULT_AP_SSID, <<"AtomVM-ESP32">>).
+-define(DEFAULT_AP_PSK, <<"esp32default">>).
```

Comment:
It would be really nice if these defaults could be set as Kconfig options with `idf.py meuconfig` rather than hard coded in the source here. But I realize it would complicate the build, so it may not be worth it.

Reply:
yeah, I think it would turn to be quite complicated.